### PR TITLE
filesio: Handle multiple file extension in output definition parser

### PR DESCRIPTION
**Pull request by ptitjano**
`parse_output_definition` only returns the default file extension even if multiple formats are supported.

This issue is fixed by parsing the `fileFilter` of the file definition. This allows to retrieve all the supported extensions. Besides, the default file format is moved at the beginning of the `supported_formats` list to ensure that it is interpreted as the default extension.

### CODE REVIEW

**Comment by ptitjano** [comment]
@dmarteau  could you take a look at it please?

File: pyqgiswps/executors/io/filesio.py [hunk 1]
```diff
@@ -128,18 +129,35 @@ def parse_output_definition(outdef: QgsProcessingOutputDefinition, kwargs,
...
-        mime = None
+        supported_formats = []
         if alg:
             inputdef = alg.parameterDefinition(outdef.name())
             if isinstance(inputdef, QgsProcessingParameterFileDestination):
-                mime = mimetypes.types_map.get("." + inputdef.defaultFileExtension())
+                default_extension = inputdef.defaultFileExtension()
+                default_extension_idx = -1
+                for filter_ in inputdef.fileFilter().split(";;"):
+                    extension_regex = QRegularExpression("[*.]([a-z]+)")
```
Comment (dmarteau) [inline pyqgiswps/executors/io/filesio.py]: Why using a `QRegularExpression` and not and  python re ?

File: pyqgiswps/executors/io/filesio.py [hunk 2]
```diff
@@ -128,18 +129,35 @@ def parse_output_definition(outdef: QgsProcessingOutputDefinition, kwargs,
...
+                    match = extension_regex.match(filter_)
+                    if match.hasMatch():
+                        extension = match.captured(1)
+                        if extension == default_extension:
+                            default_extension_idx = len(supported_formats)
+
+                        supported_formats.append(Format(match.captured(1)))
```
Comment (dmarteau) [inline pyqgiswps/executors/io/filesio.py]: You have already assigned `extension` to  `match.captured(1)`, so you may use it as `Format` argument

File: pyqgiswps/executors/io/filesio.py [hunk 3]
```diff
@@ -128,18 +129,35 @@ def parse_output_definition(outdef: QgsProcessingOutputDefinition, kwargs,
...
+
+                # The first format in the list is interpreted as the default one
+                # Ensure that it corresponds to `default_extension`
+                if default_extension_idx > 0:
+                    supported_formats.insert(0, supported_formats.pop(Format(default_extension)))
```
Comment (dmarteau) [inline pyqgiswps/executors/io/filesio.py]: This is not OGC compliant: the Format argument must be a valid mime type not a file extension.  You must use the mimetypes module to resolve file extension to valide mime types.

See https://github.com/3liz/py-qgis-wps/blob/master/pyqgiswps/inout/formats.py#L35

**Review by dmarteau — CHANGES_REQUESTED** [review]
The return format must be compliant to the OGC spec: supported formats are mime types as primary argument since they are mapped to the `MimeType` for complex output.

File: pyqgiswps/executors/io/filesio.py (hunk 1 above)
Comment (ptitjano) [reply pyqgiswps/executors/io/filesio.py]: no reason. Changed to `re`.

File: pyqgiswps/executors/io/filesio.py (hunk 3 above)
Comment (ptitjano) [reply pyqgiswps/executors/io/filesio.py]: My bad. This is fixed now.

File: pyqgiswps/executors/io/filesio.py (hunk 2 above)
Comment (ptitjano) [reply pyqgiswps/executors/io/filesio.py]: done.

**Comment by ptitjano** [comment]
> The return format must be compliant to the OGC spec: supported formats are mime types as primary argument since they are mapped to the `MimeType` for complex output.

This should be fixed now.

**Pull request merged**